packages/edit-post/src/components/sidebar/settings-header/index.js

```diff
@@ -11,11 +12,12 @@ import SidebarHeader from '../sidebar-header';
 
 const SettingsHeader = ( { openDocumentSettings, openBlockSettings, sidebarName } ) => {
 	const blockLabel = __( 'Block' );
+	const documentLabel = applyFilters( 'editor.sidebar.settingsHeader.documentLabel', __( 'Document' ) );
```

Comment:
Hi @gziolo, any thoughts related to this new filter?

Reply:
I could also use some help in the naming of this filter, if this is the right approach moving forward.

Reply:
I shared my thoughts here: https://github.com/WordPress/gutenberg/pull/17311#pullrequestreview-283376728. I marked PR as `request changes` only to ensure that we have a proper discussion about customizing UI labels with some guidelines about implementation documented before we land this change.